performance_test/src/data_running/data_runner_base.hpp

```diff
@@ -39,7 +44,28 @@ class DataRunnerBase : boost::noncopyable
 public:
   DataRunnerBase()
   : m_ec(ExperimentConfiguration::get())
-  {}
+  {
+    if (m_ec.check_memory()) {
+#ifdef MEMORY_TOOLS_ENABLED
+      osrf_testing_tools_cpp::memory_tools::initialize();
+
+      auto on_unexpected_memory =
+              [](osrf_testing_tools_cpp::memory_tools::MemoryToolsService &service) {
+                std::cerr << "unexpected malloc";
```

Comment:
This is a misleading statement. It might be `malloc`, `free`, `calloc`, or `realloc`. 

I would create separate callbacks for each call with the corresponding messages.

Reply:
This `std::cerr` should not be there. The next line already prints which type of allocation happened:

```
 malloc  (not expected) 104 -> 0x2db7440
Stack trace (most recent call last):
#33   Object "[0xffffffffffffffff]", at 0xffffffffffffffff, in 
#32   Object "/home/andreas.pasternak/1429/apex_ws/src/performance_test/performance_test/cmake-build-debug/perf_test", at 0x40a168, in 
#31   Object "/lib/x86_64-linux-gnu/libc.so.6", at 0x7f70b6cd282f, in __libc_start_main
#30   Object "/home/andreas.pasternak/1429/apex_ws/src/p
```

I removed this line. Then multiple callbacks are not required at the moment.